File: Algorithms/Linear-search/linear_search.py
n = int(input("Enter the number of elements in array: "))
a = []
print(f"Enter {n} elements")
for i in range(n):
    x = int(input())
    a.append(x)

key = int(input("Enter the key you would do search: "))

found = False
index = -1
for i in range(n):
    if a[i] == key:
        found = True
        index = i
        break
if found:
    print('Key {} found at index {}'.format(key, index))
else:
    print('Key {} not found in the array'.format(key))

File: LinearSearch.java
import java.util.Scanner;
import java.util.Random;

public class LinearSearch {
  
  public static void main(String[] args) {
    //Taking random number of instead of user input
    Random rd = new Random(); 
    int[] inputArray = new int[100];
    System.out.println("Array:");
    for (int i = 0; i < inputArray.length; i++) {
      inputArray[i] = rd.nextInt(100000) + 1; 
      System.out.print(inputArray[i] + " ");
    }
    System.out.println("\nInput Integer to Search:");
    Scanner sc = new Scanner(System.in);
    //Linear Search implementation starts here
    int searchKey = sc.nextInt();
    for (int i = 0; i < inputArray.length; i++) { 
      if (inputArray[i] == searchKey) 
        System.out.println(searchKey + " found at index: " + i);
    }
  }
  
}
